src/main/java/spark/Request.java

```diff
@@ -513,34 +513,43 @@ public String protocol() {
         return Collections.unmodifiableMap(params);
     }
 
-    private static List<String> getSplat(List<String> request, List<String> matched) {
-        int nbrOfRequestParts = request.size();
-        int nbrOfMatchedParts = matched.size();
-
-        boolean sameLength = (nbrOfRequestParts == nbrOfMatchedParts);
-
+    public static List<String> getSplat(List<String> request, List<String> matched) {
         List<String> splat = new ArrayList<>();
 
-        for (int i = 0; (i < nbrOfRequestParts) && (i < nbrOfMatchedParts); i++) {
+        int i = 0;
+        int j = 0;
+
+        while (i < matched.size() && j < request.size()) {
 
             String matchedPart = matched.get(i);
 
             if (SparkUtils.isSplat(matchedPart)) {
 
-                StringBuilder splatParam = new StringBuilder(request.get(i));
+                StringBuilder splatParam = new StringBuilder(request.get(j));
 
-                if (!sameLength && (i == (nbrOfMatchedParts - 1))) {
-                    for (int j = i + 1; j < nbrOfRequestParts; j++) {
+                if (i+1 < matched.size()) {
+                    while (j+1 < request.size() && !matched.get(i+1).equals(request.get(j+1))) {
+                        j++;
+                        splatParam.append("/");
+                        splatParam.append(request.get(j));
+                    }
+                } else {
+                    while (j+1 < request.size()) {
+                        j++;
```

Comment:
this looked like repetition. can you refactor this into a function?

Reply:
not going to save much, so will leave it as is